doc/plugin-criteria.md

```diff
@@ -0,0 +1,51 @@
+| 2. | The task provides a different solution to a problem which is also solved by kubectl. However, the task would be impossible or very hard to reproduce with kubectl alone. Examples: `access-matrix`, `get-all`, `mtail`, `tail` |  In general welcome unless it overlaps significantly with other plugins or has a bad usability.                                                                                                           |
+| 3. | The task addresses a completely different problem than any of the kubectl subcommands, but is clearly focused on k8s.                                                                                                          |  Gray area. If the task fits well into `kubectl`, it may be accepted (for example `kubectl debug`).                                                                                                       |
+| 4. | The task is very dissimilar to common kubectl tasks and is not primarily focused on k8s. However, it can be useful in the k8s ecosystem.                                                                                       | Not acceptable.                                                                                                                                                                                           |
+| 5. | The task is very dissimilar to common kubectl tasks and is not focused on k8s. Its general usefulness in the k8s ecosystem is questionable.                                                                                    | Not acceptable.                                                                                                                                                                                           |
+
+Category
+---
+
+In order of decreasing usage frequency:
+
+
+| #  | Classification                                                                                                            | Recommendation                             |
+|----|---------------------------------------------------------------------------------------------------------------------------|--------------------------------------------|
+| 1. | Recurring task on the same cluster. For example: `mtail`                                                                  | Favors acceptance as plugin.               |
+| 2. | Recurring task on different cluster instances, but one-time task on the same cluster instance. For example: `init-tiller` | Gray area. Usually this is not acceptable. |
+| 3. | One-time task per machine.                                                                                                | In general not acceptable as plugin.       |
+
+Usability
+---
+Furter required acceptance criteria:
+
+1. Standard look & feel: Adheres to standard kubectl best practices and in particular uses consistent (abbreviated) flag names. For example: bad `--ns`, good `--namespace` + `-n`.
```

Comment:
Would be good to link to a style guide here to avoid friction and nitpicking.

Reply:
Good point. @ahmetb do you know if such a document exists?

Reply:
Yeah we should probably write a separate "Style Guide for Plugins" document. We could easily fill it up.